python/shotgun_fields/shotgun_field_delegate.py

```diff
@@ -92,26 +149,201 @@ def _on_before_paint(self, widget, model_index, style_options):
+        """
+
+        # Forward mouse clicks to the underlying display widget. This only kicks
+        # in if the editor widget isn't displayed or doesn't process a mouse
+        # event for some reason. If you're having trouble with editors
+        # disappearing, it may be because they can't receive focus or aren't
+        # handling a mouse click.
+        if event.type() == QtCore.QEvent.MouseButtonRelease:
+            self._forward_mouse_event(event, index)
+            return True
+
+        return False
+
+    def _forward_mouse_event(self, mouse_event, index):
+        """
+        Forward the mouse event to the display widget to simulate
+        interacting with the widget. This is necessary since the delegate only
+        paints the widget in the view rather than being and actual widget
+        instance.
+
+        :param mouse_event: The event that occured on the delegate.
+        :type mouse_event: :class:`~PySide.QtCore.QEvent`
+        :param index: The model index that was acted on.
+        :type index: :class:`~PySide.QtCore.QModelIndex`
+        """
+
+        # get the widget used to paint this index, populate it with the
+        # value for this index
+        widget = self._get_painter_widget(index, self.view)
+        _set_widget_value(widget, index)
+
+        item_rect = self.view.visualRect(index)
+
+        # get the rect of the item in the view
+        widget.resize(item_rect.size())
+
+        # move the widget to 0, 0 so we know exactly where it is
+        widget.move(0, 0)
+
+        # map global mouse position to within item_rect
+        view_pos = self.view.viewport().mapFromGlobal(QtGui.QCursor.pos())
+
+        # calculate the offset from the item rect
+        widget_x = view_pos.x() - item_rect.x()
+        widget_y = view_pos.y() - item_rect.y()
+
+        # forward the mouse event to the display widget
+        forward_event = QtGui.QMouseEvent(
+            mouse_event.type(),
+            QtCore.QPoint(widget_x, widget_y),
+            mouse_event.button(),
+            mouse_event.buttons(),
+            mouse_event.modifiers(),
+        )
+        QtGui.QApplication.sendEvent(widget, forward_event)
+
+
+def _set_widget_value(widget, model_index):
+    """
+    Updates the supplied widget with data from the supplied model index.
+
+    :param widget: The widget to set the value for
+    :type parent: :class:`~PySide.QtGui.QWidget`
+    :param model_index: The index of the model where the data comes from
+    :type model_index: :class:`~PySide.QtCore.QModelIndex`
+    """
+
+    src_index = _map_to_source(model_index)
+    if not src_index or not src_index.isValid():
+        # invalide index, do nothing
+        return
+
+    # special case for image fields
+    if widget._field_name == "image":
+        primary_item = src_index.model().item(src_index.row(), 0)
+        icon = primary_item.icon()
+        if icon:
+            widget.set_value(icon.pixmap(QtCore.QSize(256, 256)))
```

Comment:
Yes, probably. Let me figure out what SG does as far as limiting thumbnails and I'll update.